contracts/SwapAndCall.sol

```diff
@@ -0,0 +1,80 @@
+pragma solidity ^0.5.0;
+
+import '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import '@openzeppelin/contracts/token/ERC20/SafeERC20.sol';
+import 'hardlydifficult-ethereum-contracts/contracts/proxies/CallContract.sol';
+
+
+/**
+ * @notice call any contract(s) and then refund the remaining balance (if any)
+ * Idea from https://etherscan.io/address/0x11111254369792b2Ca5d084aB5eEA397cA8fa48B#code
+ */
+contract SwapAndCall
+{
+  using CallContract for address;
+  using SafeERC20 for IERC20;
+
+  // Accept ETH from other contracts
+  function() external payable {}
+
+  /**
+   * @notice Make a series of calls and then refunds any funds remaining.
+   * @param _contractsAndTokensToRefund The contract address for each call to make.
+   * Followed by the address of any ERC-20 tokens to refund at the end.
+   * ETH is always refunded.  If no additional token is required this may be empty.
+   * @param _callDataConcat Call data for each call, appended into a single bytes field.
+   * @param _startPositionsAndValues The start position of callData for each call after
+   * the first (assumed to be 0).
+   * Followed by the call value to include with each call.
+   * @dev Variables are reused to avoid a `stack too deep` error.
+   */
+  function swapAndCall(
+    address[] calldata _contractsAndTokensToRefund,
+    bytes calldata _callDataConcat,
+    uint[] calldata _startPositionsAndValues
+  ) external payable
+  {
+    uint callCountMinusOne = _startPositionsAndValues.length / 2;
```

Comment:
The naming of this confuses me, but I'm certain you have a good reason for it;)
If there are 4 calls, + the associated `Value` for each call, that would mean 8 items in the array. in this case, 
wouldn't `callCountMinusOne` evaluate to 4? (the naming would lead me to believe it should be `3`)

Reply:
The first position is assumed to be 0.  so it's not included.

Reply:
Right...just like your comment on the param stated clearly. sorry.